packages/entities/entities-plugins/src/components/PluginCustomGrid.vue

```diff
@@ -0,0 +1,274 @@
+            :config="config"
+            :no-route-change="noRouteChange"
+            :plugin="plugin"
+            @custom-plugin-delete="() => handleCustomPluginDelete(plugin)"
+            @plugin-clicked="emitPluginData"
+          />
+        </div>
+      </template>
+
+      <div class="plugin-card-container">
+        <PluginSelectCard
+          v-for="(plugin, index) in getPluginCards('hidden')"
+          :key="index"
+          :can-delete-custom="canDeleteCustom"
+          :can-edit-custom="canEditCustom"
+          :config="config"
+          :no-route-change="noRouteChange"
+          :plugin="plugin"
+          @plugin-clicked="emitPluginData"
+        />
+      </div>
+    </KCollapse>
+
+    <DeleteCustomPluginSchemaModal
+      v-if="openDeleteModal && selectedPlugin"
+      :plugin="selectedPlugin"
+      @closed="handleClose"
+      @proceed="() => handleClose(true)"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, type PropType } from 'vue'
+import {
+  PluginGroup,
+  type KongManagerPluginFormConfig,
+  type KonnectPluginFormConfig,
+  type PluginType,
+  type PluginCardList,
+} from '../types'
+import composables from '../composables'
+import PluginSelectCard from './PluginSelectCard.vue'
+import DeleteCustomPluginSchemaModal from './DeleteCustomPluginSchemaModal.vue'
+
+const props = defineProps({
+  /** The base konnect or kongManger config. Pass additional config props in the shared entity component as needed. */
+  config: {
+    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    required: true,
+    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+      if (!config || !['konnect', 'kongManager'].includes(config?.app)) return false
+      if (!config.getCreateRoute) return false
+      return true
+    },
+  },
+  /**
+   * Whether or not user has rights to create custom plugins
+   */
+  canCreateCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Whether or not user has rights to delete custom plugins
+   */
+  canDeleteCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Whether or not user has rights to edit custom plugins
+   */
+  canEditCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Plugins to display in the grid
+   */
+  pluginList: {
+    type: Object as PropType<PluginCardList>,
+    default: () => ({}),
+  },
+  /**
+   * @param {boolean} noRouteChange if true, let consuming component handle event when clicking on a plugin
+   * Used in conjunction with `@plugin-clicked` event
+   */
+  noRouteChange: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Number of plugins to always have visible (never will be collapsed)
+   */
+  pluginsPerRow: {
+    type: Number,
+    default: 4,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'plugin-clicked', plugin: PluginType): void,
+  (e: 'revalidate'): void,
```

Comment:
question: what is this `revalidate` event used for?

Also, I don't see it in the docs

Reply:
It's internal only (used in `PluginSelect.vue`), we currently aren't doc'ing this component because it isn't exposed for consumption. You have the ability to delete custom plugins from the select page which means we need to refresh the plugins list afterwards.

Reply:
Can you just add a note that the revalidate event is internal-only?